snooty/postprocess.py

```diff
@@ -615,6 +616,59 @@ def add_titles_to_label_targets(
                     target.children = heading.children
             self.pending_targets = []
 
+    def bound_included_AST(
+        self,
+        nodes: MutableSequence[n.Node],
+        start_after_text: str,
+        end_before_text: str,
+    ) -> Tuple[MutableSequence[n.Node], bool, bool]:
+        """Given an AST in the form of nodes, return a subgraph of that AST by removing nodes 'outside' of 
+        the bound formed by the nodes containing the start_after_text or end_before_text. In in-order traversal,
+        a node is considered 'outside' the subgraph if it precedes and is not any ancestor of the start-after node,
+        or if it succeeds and is not any ancestor of the end-before node."""
+
+        def is_bound(node: n.Node, search_text: str) -> bool:
+            """Helper function to determine if the given node contains specified start-after or end-before text. 
+            
+            Note: For now, we are only splicing included files based on Text and TargetIdentifier nodes. 
+            Comments have Text nodes as children; Labels have TargetIdentifiers as children. 
+            So, writers can splice on these parent directives as well."""
+            if isinstance(node, n.Text) and node.value:
+                return search_text == node.value
+            elif isinstance(node, n.TargetIdentifier):
+                if node.ids and isinstance(node.ids[0], str):
+                    return search_text in node.ids
```

Comment:
It seems like it may be clearer to search for a `Comment` or `Label` class, and then dive into their children to check for a matching string. This would avoid evaluating many `Text`/`TargetIdentifier` nodes that are just red herrings, and feels more along the lines of what a reader would expect to see here given the criteria for start-after/end-before.